File: FinanceApp.Shared/Dto/Income/CreateIncome.cs
using FinanceApp.Shared.Dto.Base;

namespace FinanceApp.Shared.Models
{
    public class CreateIncome : CreateDto
    {
        public string Name { get; set; }
        public int Type { get; set; }
        public decimal Amount { get; set; }
        public DateTime InitialDate { get; set; }
        public int Frequency { get; set; }
        public DateTime? EndDate { get; set; }
        public bool IsEndless { get; set; }
        public int? TimesRecurrence { get; set; }

    }
}

File: FlutterApp/finance_app/lib/models/income/create_income.dart
class CreateIncome {
  String name;
  int recurrence;
  double amount;
  DateTime initialDate;
  DateTime? endDate;
  bool isEndless;
  int timesRecurrence;

  CreateIncome(
      {required this.name,
      required this.recurrence,
      required this.amount,
      required this.initialDate,
      required this.endDate,
      required this.isEndless,
      required this.timesRecurrence});

  Map<String, dynamic> toJson() => {
        'name': name,
        'recurrence': recurrence,
        'amount': amount,
        'initialDate': initialDate,
        'endDate': endDate,
        'isEndless': isEndless,
        'timesRecurrence': timesRecurrence,
      };
}
